plugins/module_utils/template.py

```diff
@@ -320,26 +346,34 @@ def get_template_object_name_by_uuid(self, object_type, uuid, fail_module=True):
         if response_object:
             return response_object.get("name")
 
-    def get_template_object_by_uuid(self, object_type, uuid, fail_module=True):
+    def get_template_object_by_uuid(self, object_type, uuid, fail_module=True, use_cache=False):
         """
         Retrieve a specific object type in the MSO template using its UUID.
-        :param object_type: The type of the object to retrieve the name for -> Str
-        :param uuid: The UUID of the object to retrieve the name for -> Str
+        :param object_type: The type of the object to retrieve -> Str
+        :param uuid: The UUID of the object to retrieve -> Str
+        :param use_cache: Use the cached result of the templates/objects API for the UUID -> Bool
         :return: Dict | None: The processed result which could be:
             When the UUID is existing, returns object -> Dict
             When the UUID is not existing -> None
         """
-        response_object = self.mso.request("templates/objects?type={0}&uuid={1}".format(object_type, uuid), "GET")
+        response_object = None
+        if use_cache and uuid in self.template_objects_cache.keys():
```

Comment:
Can we just do away with use_cache?
```
if uuid in self.template_objects_cache.keys():
    response_object = self.template_objects_cache[uuid]
```

Reply:
I would need to move `if use_cache:` down to line 364 where it adds the object to the cache instead.
Or do you think we can remove `use_cache` and always cache? I was thinking of making it optional at first so it doesn't create any unexpected side effects in other modules.